impl/src/main/java/org/ehcache/internal/store/DefaultStoreProvider.java

```diff
@@ -40,6 +41,7 @@
 /**
  * @author Ludovic Orban
  */
+@SupplementaryService
```

Comment:
If both are available, the transactional one will take precedence but its implementation makes it delegate to DefaultStoreProvider when a cache doesn't have a transactional config.